### DISCUSSION TITLE
Why is this engine not default?

### DISCUSSION THREAD

**Discussion by Gamepro5** [godot-jolt/godot-jolt#531] [General]
I'd like to speak of this from the perspective of a user who simply needs one thing from a physics engine: Reliable collision detection.

As such, I am likely ignorant to the bigger picture, but I would like to know why Jolt hasn't replaced Godot Physics. When I was using Godot Physics, I was in constant contact with one of the last remaining devs for it and all the bugs I found just sat unfixed for years. They need more manpower.

You seem to be one man carrying this entire project, and I don't know if you are getting paid or not, but judging by the fact that this is a free addon, I am kinda baffled as to why the main Godot team hasn't just asked you to join them and incorporate jolt into godot's core.

Im sure there is a reason, and I would love to know what it is. In the meantime, I plan to continue using Jolt and will 100% be donating if and when my game makes any money.

Cheers!

**Comment by mihe** [comment]
I appreciate the support and kind words.

> You seem to be one man carrying this entire project

While I might be the only contributor to this particular extension, I've received a lot of generous support from the author of Jolt as well, so I wouldn't say I'm the only one carrying this project.

> I don't know if you are getting paid or not

I'm not, no, apart from the donations I get through GitHub Sponsors, but those only just barely cover the cost for the code signing certificates. Making money from this was never the goal though. This project mostly started out as something that I could add to my resume, to wrap up the hiatus that I've been on for the past two years now, but I must admit it got more traction (and took longer) than I expected, so I've pushed that wrap-up in front of me for a few months now. I will likely start my return to the real world within the next few weeks though, which will significantly cut down on the amount of time I can spend on this.

>  I am kinda baffled as to why the main Godot team hasn't just asked you to join them and incorporate jolt into godot's core.

Nothing is certain yet, but there is at least a non-zero chance that this will happen [at some point](https://twitter.com/reduzio/status/1680705434729021441) in the future, especially now that Jolt has support for [soft-body physics](https://twitter.com/jrouwe/status/1687051655898955776).

There is also [an active proposal](https://github.com/godotengine/godot-proposals/issues/7308) that's been put forward by Godot's production team, as an intermediate step, for recognizing Godot Jolt as an official add-on for Godot. What that will mean in the near-term isn't clear to me, but it seems like they want to make it easy to install a small subset of extensions through some yet-to-be-implemented functionality in the editor.
